OpenRA.Mods.Common/Traits/Render/WithTextDecoration.cs

```diff
@@ -0,0 +1,120 @@
+﻿#region Copyright & License Information
+/*
+ * Copyright 2007-2016 The OpenRA Developers (see AUTHORS)
+ * This file is part of OpenRA, which is free software. It is made
+ * available to you under the terms of the GNU General Public License
+ * as published by the Free Software Foundation, either version 3 of
+ * the License, or (at your option) any later version. For more
+ * information, see COPYING.
+ */
+#endregion
+
+using System;
+using System.Collections.Generic;
+using System.Drawing;
+using System.Linq;
+using OpenRA.Graphics;
+using OpenRA.Mods.Common.Graphics;
+using OpenRA.Traits;
+
+namespace OpenRA.Mods.Common.Traits.Render
+{
+	[Desc("Displays a text overlay relative to the selection box.")]
+	public class WithTextDecorationInfo : UpgradableTraitInfo
+	{
+		[FieldLoader.Require] [Translate] public readonly string Text = null;
+
+		public readonly string Font = "TinyBold";
+
+		public readonly Color Color = Color.White;
+
+		[Desc("Point in the actor's selection box used as reference for offsetting the decoration image. " +
+			"Possible values are combinations of Center, Top, Bottom, Left, Right.")]
+		public readonly ReferencePoints ReferencePoint = ReferencePoints.Top | ReferencePoints.Left;
+
+		[Desc("The Z offset to apply when rendering this decoration.")]
+		public readonly int ZOffset = 1;
+
+		[Desc("Player stances who can view the decoration.")]
+		public readonly Stance Stances = Stance.Ally;
```

Comment:
I recently fixed `GivesBounty` to use `ValidStances` so  now we have:
```
tmba:openra thill $ ag "Stance Stances" --csharp | wc -l
       1
tmba:openra thill $ ag "Stance\[\] Stances" --csharp | wc -l
       1
tmba:openra thill $ ag "Stance ValidStances" --csharp | wc -l
       4
```

So I think it'd only do us injustice to introduce another `Stances` instead of fixing them both.